File: backup/labs/compliance-portal/app/backend/app/privacy_engine/consent_analyzer.py
class ConsentAnalyzer:
    def analyze(self, consents: list = None) -> dict:
        consents = consents or []
        total = len(consents) or 1
        approved = len([c for c in consents if c["status"] == "Approved"])
        pending = len([c for c in consents if c["status"] == "Pending"])
        revoked = len([c for c in consents if c["status"] == "Revoked"])

        coverage = round((approved / total) * 100)

        return {
            "active": approved,
            "pending": pending,
            "revoked": revoked,
            "total": len(consents),
            "consentCoverage": coverage
        }

consent_analyzer = ConsentAnalyzer()

File: backup/labs/compliance-portal/app/server/analysis/consentAnalyzer.js
import { analyseCompliance } from "../privacy/scorer.js";

// Consent totals, tolerant of both source shapes: the e-commerce API reports
// `consent_status: granted|revoked`, the finance API the same field, and the
// legacy EverShop path maps to Approved/Revoked/Pending.
class ConsentAnalyzer {
  async analyze(consents = [], sector = "generic") {
    const { totals, percentages } = analyseCompliance(consents, sector);

    return {
      active: totals.granted,
      pending: totals.pending,
      revoked: totals.revoked,
      total: totals.total,
      consentCoverage: percentages.granted,
      revocationRate: percentages.revoked
    };
  }
}

export default new ConsentAnalyzer();
